### ISSUE TITLE
How to set the parameters for hacnn?

### ISSUE THREAD

**Issue by nlj0011** [KaiyangZhou/deep-person-reid#27]
Thank you for providing the code.I learn a lot from it. However, I meet some problem when train the HACNN with xent loss, I set 120 epoch, and the parameters are:
--height 160 --width 64 --max-epoch 120-train-batch 32 --test-batch 32 --stepsize 20 --eval-step 20
However, the Rank1 result on Market-1501 is only 82.1% with the mAP 61.2%, which are worsing than your providing results. When I try the hacnn with xent+htri, the results in epoch150 are only 57.2% Rank-1 with 35.6% mAP. I don't know whether the hyperparameters are set right. Could you teach me how to set the parameter when you train the hacnn?
Thanks!

**Comment by KaiyangZhou** [comment]
HACNN is trained from scratch so more training epochs are preferred. In terms of optim algorithms, sgd usually outperforms adaptive methods like adam on training models from scratch as adaptive optim methods would make model stuck in local optimum while sgd is better at escaping those local optima. You may try using sgd with max-epoch=300 and lr starting from 0.03. Decay the lr by 0.1 after 50% and 75% of training time. With these settings you could achieve around rank1=91 on market.

**Comment by nlj0011** [comment]
Thanks for your reply~I will try it again, best regards!

**Comment by tomFoxxxx** [comment]
> Thank you for providing the code.I learn a lot from it. However, I meet some problem when train the HACNN with xent loss, I set 120 epoch, and the parameters are:
> --height 160 --width 64 --max-epoch 120-train-batch 32 --test-batch 32 --stepsize 20 --eval-step 20
> However, the Rank1 result on Market-1501 is only 82.1% with the mAP 61.2%, which are worsing than your providing results. When I try the hacnn with xent+htri, the results in epoch150 are only 57.2% Rank-1 with 35.6% mAP. I don't know whether the hyperparameters are set right. Could you teach me how to set the parameter when you train the hacnn?
> Thanks!

hi   I wanna what's the xcent loss, do u have any reference material? websites or something else.
Thanks

**Comment by sakura811** [comment]
> HACNN is trained from scratch so more training epochs are preferred. In terms of optim algorithms, sgd usually outperforms adaptive methods like adam on training models from scratch as adaptive optim methods would make model stuck in local optimum while sgd is better at escaping those local optima. You may try using sgd with max-epoch=300 and lr starting from 0.03. Decay the lr by 0.1 after 50% and 75% of training time. With these settings you could achieve around rank1=91 on market.

Hi Kaiyang, I trained the model with your HACNN and the same parematers on market1501, but the final results mAP=68%, Rank1=85.6%. I could not get the Rank1=91%. 
So I want to konw **if you make an data argumentation on market1501 for HACNN**, like you mentioned in OSNet paper ?
Thank you very much.
